README.md

```diff
@@ -1,2 +1,308 @@
+
+#### Return Value ####
+
+None.
+
+#### Example ####
+
+```squirrel
+// Sets the charge voltage to 4200mV
+batteryCharger.setChargeVoltage(4200);
+```
+
+
+
+### getChargeVoltage() ###
+
+Returns the target battery charge voltage
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+
+local voltage = batteryCharger.getChargeVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+
+### getBatteryVoltage() ###
+
+Returns the current battery voltage based on the internal ADC conversion.
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getBatteryVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getVBUSVoltage() ###
+
+Returns the VBUS voltage based on the ADC conversion, this is the input voltage
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getVBUSVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getSystemVoltage() ###
+
+Returns the system voltage based on the ADC conversion, this the output voltage which can be used to power other chips in your application
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getSystemVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getChargingCurrent() ###
+
+Returns the measured current going to the battery 
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The charging current in milli Amps.
+
+#### Example ####
+
+```squirrel
+local current = batteryCharger.getChargingCurrent();
+server.log("Current: " + current + "mA");
+```
+### getChargingStatus() ###
+
+ Returns the charging status
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — Not Charging = 0,  Pre-charge = 1, Fast Charging = 2, Charge Termination Good = 3
+```squirrel
+enum BQ25895M_CHARGING_STATUS{
+    NOT_CHARGING = 0x00, // 0
+    PRE_CHARGE = 0x08, // 1
+    FAST_CHARGING = 0x10, // 2
+    CHARGE_TERMINATION_DONE = 0x18 // 3
+}
+```
+#### Example ####
+
+```squirrel
+local status = charger. getChargingStatus();  
+switch(status) {  
+	case BQ25895M_CHARGING_STATUS.NOT_CHARGING :  
+		// Do something  
+		break;  
+	case BQ25895M_CHARGING_STATUS.PRE_CHARGE :  
+		// Do something  
+		break;  
+	case BQ25895M_CHARGING_STATUS.FAST_CHARGING :  
+		// Do something  
+		break;  
+	case BQ25895M_CHARGING_STATUS.CHARGE_TERMINATION_DONE :  
+		// Do something  
+		break;  
+}
+```
+### getChargerFaults() ###
+
+ Returns the possible charger faults
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+Table &mdash; the possible charger faults
+
```

Comment:
Add sub header here to introduce this table. Update enums from code to tables also with headings. 

##### Charger Faults #####

(fault table here)

##### Charging Fault #####

(update enum to table here)

##### NTC Fault #####

(update enum to table here)